test/envoye2e/stackdriver_plugin/fake_stackdriver/fake_stackdriver.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fakestackdriver
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net"
+
+	grpc "google.golang.org/grpc"
+
+	empty "github.com/golang/protobuf/ptypes/empty"
+	"google.golang.org/genproto/googleapis/api/metric"
+	"google.golang.org/genproto/googleapis/api/monitoredres"
+	monitoringpb "google.golang.org/genproto/googleapis/monitoring/v3"
+)
+
+// FakeStackdriverServer is a fake stackdriver server which implements all of monitoring v3 service method.
+type FakeStackdriverServer struct {
```

Comment:
could you just have:

```golang
type FakeSDServer struct {
  monitoringpb.MetricsServiceServer
  
  RcvReq chan *monitoringpb.CreateTimeSeriesRequest
}

// CreateTimeSeries implements CreateTimeSeries method.
func (s *FakeStackdriverServer) CreateTimeSeries(ctx context.Context, req *monitoringpb.CreateTimeSeriesRequest) (*empty.Empty, error) {
	s.RcvReq <- req
	return &empty.Empty{}, nil
}

```

instead of having all of the empty interface methods here?

Reply:
It won't work:
```
cannot use fsds (type *FakeStackdriverServer) as type monitoring.MetricServiceServer in argument to monitoring.RegisterMetricServiceServer:
	*FakeStackdriverServer does not implement monitoring.MetricServiceServer (missing CreateMetricDescriptor method)
```